include/System/MarioGamePad.hpp

```diff
@@ -1,20 +1,306 @@
+		MEANING_0x2      = 0x2,
+		MEANING_0x4      = 0x4,
+		MEANING_0x8      = 0x8,
+		MEANING_0x10     = 0x10,
+		MEANING_0x20     = 0x20,
+		MEANING_0x40     = 0x40,
+		MEANING_0x80     = 0x80,
+		MEANING_0x100    = 0x100,
+		MEANING_0x200    = 0x200, // Unused?
+		MEANING_0x400    = 0x400,
+		MEANING_0x800    = 0x800,
+		MEANING_0x1000   = 0x1000,
+		MEANING_0x2000   = 0x2000,
+		MEANING_0x4000   = 0x4000,
+		MEANING_0x8000   = 0x8000,
+		MEANING_0x10000  = 0x10000,
+		MEANING_0x20000  = 0x20000,
+		MEANING_0x40000  = 0x40000,
+		MEANING_0x80000  = 0x80000,
+		MEANING_0x100000 = 0x100000,
+		MEANING_0x200000 = 0x200000,
+	};
+	enum PadFlags {
+		PAD_FLAG_0x1  = 0x1,
+		PAD_FLAG_0x2  = 0x2,
+		PAD_FLAG_0x4  = 0x4,
+		PAD_FLAG_0x8  = 0x8,
+		PAD_FLAG_0x10 = 0x10,
+		PAD_FLAG_0x80 = 0x80,
+	};
+
+	struct StickPos {
+		f32 x;
+		f32 y;
+	};
+
+	// Fabricated
+	inline bool checkReset(s32* resetPort)
+	{
+		if (resetPort != 0) {
+			*resetPort = JUTGamePad::C3ButtonReset::sResetOccurredPort;
+		}
+		return JUTGamePad::C3ButtonReset::sResetOccurred;
+	}
+
+	// Fabricated
+	inline void handleReset(s32 resetPort)
+	{
+		if (resetPort == JUTGamePad::EPortInvalid) {
+			mResetFlag |= 0xf;
+		} else {
+			mResetFlag |= (u16)(1 << resetPort);
+		}
+		JUTGamePad::C3ButtonReset::sResetOccurred = false;
+	}
+
+	// Fabricated
+	inline void updateMeaning(EButtons button, PadMeanings padButton,
+	                          u32 prevMeaning)
+	{
+		if ((mButton.mTrigger & button) != 0
+		    || ((mButton.mButton & button) != 0
+		        && ((prevMeaning & padButton) != 0))) {
+			mMeaning |= padButton;
+		}
+	}
+
+	// Fabricated
+	inline void resetStick(StickPos& stick)
+	{
+		stick.x = 0.0f;
+		stick.y = 0.0f;
+	}
+
+	// Fabricated
+	inline u32 resetMeaning()
+	{
+		u16 dc    = this->_DC;
+		this->_DC = 0;
+
+		if (this->mButton.mButton
+		    & (DPAD_UP | DPAD_DOWN | DPAD_LEFT | DPAD_RIGHT)) {
+			this->_DC |= 1;
+		}
+
+		if ((dc & 1) != 0) {
+			if (this->mButton.mAnalogRf > 0.5f) {
+				this->_DC |= 1;
+			}
+		} else if (this->mButton.mAnalogRf > 0.25f) {
+			this->_DC |= 1;
+		}
+
+		this->_DE = this->_DC & ~dc;
+		this->_E0 = dc & ~this->_DC;
+
+		for (int i = 0; i < VARIANTS; i++) {
+			resetStick(this->mCompSPos[i]);
+		}
+
+		u32 prevMeaning = this->mMeaning;
+		this->mMeaning  = 0;
+		return prevMeaning;
+	}
+
+	// Fabricated
+	inline void updateMeaningLCamera(u32 prevMeaning)
+	{
+		// This is for when in y camera
+		this->mCompSPos[1].x = (f32)this->mButton.mAnalogL;
+		this->mCompSPos[1].y = (f32)this->mButton.mAnalogR;
+
+		this->updateMeaning(R, MEANING_0x400, prevMeaning);
+		this->updateMeaning(Z, MEANING_0x1000, prevMeaning);
+		this->updateMeaning(L, MEANING_0x2000, prevMeaning);
+		this->mCompSPos[2].x = this->mMainStick.mPosX;
+		this->mCompSPos[2].y = this->mMainStick.mPosY;
+		this->updateMeaning(A, MEANING_0x10000, prevMeaning);
+		this->updateMeaning(B, MEANING_0x10000, prevMeaning);
+		this->updateMeaning(Y, MEANING_0x4000, prevMeaning);
```

Comment:
I haven't used `this->` prefix in most of the code, so I think we should keep following that convention. I'll write codestyle guidelines at some point to fix stuff like this in stone